core/web/transfer_controller.go

```diff
@@ -40,9 +40,24 @@ func (tc *TransfersController) Create(c *gin.Context) {
 		return
 	}
 
+	if tr.FromAddress == utils.ZeroAddress {
+		jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("invalid withdrawal source address address: %v", tr.FromAddress))
+		return
+	}
+
+	if !tr.AllowHigherAmounts {
+		balance := chain.BalanceMonitor().GetEthBalance(tr.FromAddress)
+
+		// ETH balance is less than the sent amount
+		if balance == nil || balance.Cmp(&tr.Amount) == -1 {
+			jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("balance is too low for this transaction to be executed: %v", balance))
+			return
+		}
+	}
+
 	db := tc.App.GetSqlxDB()
 	q := pg.NewQ(db, tc.App.GetLogger(), tc.App.GetConfig())
-	etx, err := bulletprooftxmanager.SendEther(q, chain.ID(), tr.FromAddress, tr.DestinationAddress, tr.Amount, chain.Config().EvmGasLimitTransfer())
+	etx, err := chain.TxManager().SendEther(q, chain.ID(), tr.FromAddress, tr.DestinationAddress, tr.Amount, chain.Config().EvmGasLimitTransfer())
```

Comment:
It is no longer necessary to pass in the `q` because the bptxm has one attached to itself.